lib/internal/modules/cjs/loader.js

```diff
@@ -562,6 +562,21 @@ Module._resolveFilename = function(request, parent, isMain, options) {
     return request;
   }
 
+  const parentFilename = parent && parent.filename;
+  const cache = Module._resolveFilename.cache;
+
+  let filename;
+  // do not use the cache
+  // 1. require.resolve('you/request', {paths: []})
+  // 2. don't have parent.filename
+  // 3. don't have cache for this parent.filename
```

Comment:
Comments here is a little bit confusing. It's better to say `"Using the cache when ... "`